hameed_hw/index.js

```diff
@@ -0,0 +1,89 @@
+// Create an http server that can add and subtract from a number, which we will call the "state".
+// To run this file: node .
+// Then in your browser: http://localhost:3000
+let http = require('http');
+// declare the state variabel which start from 10
+let state = 10
+let server = http.createServer();
+// give the port a number
+const port = 3000;
+//To listen to the port and console the state of this port 
+server.listen(port, function(error) {
+  if (error) {
+    console.log(error);
+  } else {
+    console.log('api listening on port', port);
+  }
+});
+// creat a response for the request url
+server.on('request', function(request, response) {
+  //console.log(response);
+  console.log('New http request received', request.url);
+  //use switch to determine the request url case and prosses the resultes
+  switch (request.url) {
+  	// /state 
+	// response: the current state in a html format 
+	// when the server starts, this should return "10"
+  	//in this case we just show the curent state value 
+  case '/state':
+    console.log('state is ' + state);
+    showState(state);
+    break;
+	// This should add 1 to the current state
+  case '/add':
+     state++;
+    console.log('state is ' + state);
+    response.setHeader('content-type', 'text/html');
+ 	showState(state);
+    break;
+    // This should subtract 1 ƒrom the current state
+  case '/remove':
+     state--;
+    console.log('state is ' + state);
+    showState(state);
+    break;
+    // This should set the state back to 10
+     // I didnt want to start with error so I did the first step as reset and show the state.
+  case '/reset':
```

Comment:
What will happen here if the `/reset` case is hit? What lines of code then get executed?